tests/test_sequence_interpolator.py

```diff
@@ -0,0 +1,429 @@
+import numpy as np
+import pytest
+
+from experanto.interpolators import (
+    PhaseShiftedSequenceInterpolator,
+    SequenceInterpolator,
+)
+
+from .create_sequence_data import sequence_data_and_interpolator
+
+
+@pytest.mark.parametrize("n_signals", [0, 1, 10, 50])
+@pytest.mark.parametrize("sampling_rate", [3.0, 10.0, 100.0])
+@pytest.mark.parametrize("use_mem_mapped", [False, True])
+def test_nearest_neighbor_interpolation(n_signals, sampling_rate, use_mem_mapped):
+    with sequence_data_and_interpolator(
+        data_kwargs=dict(
+            n_signals=n_signals,
+            use_mem_mapped=use_mem_mapped,
+            t_end=5.0,
+            sampling_rate=sampling_rate,
+        )
+    ) as (timestamps, data, _, seq_interp):
+        assert isinstance(
+            seq_interp, SequenceInterpolator
+        ), "Interpolation object is not a SequenceInterpolator"
+        assert not isinstance(
+            seq_interp, PhaseShiftedSequenceInterpolator
+        ), "Interpolation object is a PhaseShiftedSequenceInterpolator"
+
+        times = timestamps[:10] + 1e-9
```

Comment:
@reneburghardt tbh, I am not sure if this epsilon is needed here. Have you tried without it?